cardinal/system/system_meat_puppet_system.go

```diff
@@ -0,0 +1,319 @@
+package system
+
+import (
+	"fmt"
+
+	"pkg.world.dev/world-engine/cardinal"
+	"pkg.world.dev/world-engine/cardinal/message"
+	"pkg.world.dev/world-engine/cardinal/search/filter"
+	"pkg.world.dev/world-engine/cardinal/types"
+
+	"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/component"
+	"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/constants"
+	"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/enums"
+	"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/msg"
+)
+
+// Define a variable to hold the TokeniserSystem instance
+var ts *TokeniserSystem
+
+func ProcessCommandsTokens(world cardinal.WorldContext) error {
+	return cardinal.EachMessage[msg.ProcessCommandsMsg, msg.ProcessCommandsReply](
+		world,
+		func(messageData message.TxData[msg.ProcessCommandsMsg]) (msg.ProcessCommandsReply, error) {
+			playerEntity, err := findExistingPlayer(world, messageData.Msg.PlayerName)
+			if err != nil {
+				if isDevelopmentMode() {
+					logger.Errorf("\033[31mError searching for Player entity: %v\033[0m", err)
+				}
+
+				return msg.ProcessCommandsReply{
+					Success: false,
+					Message: fmt.Sprintf("Error searching for Player entity: %v", err),
+				}, err
+			}
+
+			player, err := getPlayerEntity(world, playerEntity)
+			if err != nil {
+				if isDevelopmentMode() {
+					logger.Errorf("\033[31mError getting Player: %v\033[0m", err)
+				}
+
+				return msg.ProcessCommandsReply{
+					Success: false,
+					Message: fmt.Sprintf("Error getting Player: %v", err),
+				}, err
+			}
+
+			// variables to use
+			pID := player.PlayerID
+			// rID := player.RoomID ---> Not used YET
+			var tokens = messageData.Msg.Tokens
+			if isDevelopmentMode() {
+				logger.Infof("\033[33mTokens are: %v\033[0m", tokens)
+			}
+			var er uint8
+			var move bool
+			//var nxt uint32 ---> Not used YET
+
+			ts = NewTokeniserSystem()
+
+			if uint8(len(tokens)) > constants.MAX_TOK { // Check length of the tokens againts the max number.
+				er = constants.ER_PR_TK_CX
+			}
+
+			var tok1 string
+			tok1 = tokens[0]
+			if isDevelopmentMode() {
+				logger.Debugf("\033[35m---->CMD: %s\033[0m", tok1)
+			}
+			tokD := ts.GetDirectionType(tok1)
+
+			if tokD != enums.DirectionTypeNone {
```

Comment:
this should be moved into a switch statement its a massive if else in solidity because it has no switch

Reply:
really it has to do this, is it a movement form -> movement, is it a look -> look, otherwise go to everything else